org.eclipse.scout.rt.oauth2/src/main/java/org/eclipse/scout/rt/oauth2/OAuth2Helper.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2010-2022 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.oauth2;
+
+import java.io.IOException;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.exception.ProcessingException;
+import org.eclipse.scout.rt.platform.util.date.IDateProvider;
+
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+@ApplicationScoped
+public class OAuth2Helper {
+
+  protected final ConcurrentHashMap<OAuth2Config, TokenEntry> m_tokenCache = new ConcurrentHashMap<>();
+  protected final IDateProvider m_dateProvider = BEANS.get(IDateProvider.class);
```

Comment:
Problematic approach to have a member variable within an application scoped bean, especially one to be design for replacement within a test. If a test replaces the IDateProvider, but OAuth2Helper was instantiated before, m_dateProvider still points to the original implementation. Hard to find test failures :-) Even if called quite often here, just online BEANS.get call to avoid this issue